app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionPanelView.kt

```diff
@@ -0,0 +1,217 @@
+            if (bucketedTrackers[CROSS_SITE_TRACKING_COOKIES]?.isNotEmpty() == true)
+                View.VISIBLE else View.GONE
+        social_media_trackers.visibility =
+            if (bucketedTrackers[SOCIAL_MEDIA_TRACKERS]?.isNotEmpty() == true)
+                View.VISIBLE else View.GONE
+        fingerprinters.visibility = if (bucketedTrackers[FINGERPRINTERS]?.isNotEmpty() == true)
+            View.VISIBLE else View.GONE
+        tracking_content.visibility = if (bucketedTrackers[TRACKING_CONTENT]?.isNotEmpty() == true)
+            View.VISIBLE else View.GONE
+        cryptominers.visibility = if (bucketedTrackers[CRYPTOMINERS]?.isNotEmpty() == true)
+            View.VISIBLE else View.GONE
+    }
+
+    private fun setCategoryClickListeners() {
+        social_media_trackers.setOnClickListener {
+            interactor.openDetails(SOCIAL_MEDIA_TRACKERS)
+        }
+        fingerprinters.setOnClickListener {
+            interactor.openDetails(FINGERPRINTERS)
+        }
+        cross_site_tracking.setOnClickListener {
+            interactor.openDetails(CROSS_SITE_TRACKING_COOKIES)
+        }
+        tracking_content.setOnClickListener {
+            interactor.openDetails(TRACKING_CONTENT)
+        }
+        cryptominers.setOnClickListener {
+            interactor.openDetails(CRYPTOMINERS)
+        }
+    }
+
+    private fun setUIForDetailsMode(
+        category: TrackingProtectionCategory
+    ) {
+        normal_mode.visibility = View.GONE
+        details_mode.visibility = View.VISIBLE
+        category_title.text = context.getString(category.title)
+        val stringList = bucketedTrackers[category]?.joinToString("\n")
+        blocking_text_list.text = stringList
+        category_description.text = context.getString(category.description)
+        details_back.setOnClickListener {
+            interactor.onBackPressed()
+        }
+    }
+
+    private fun getHashMapOfTrackersForCategory(
+        list: List<Tracker>
+    ): HashMap<TrackingProtectionCategory, List<String>> {
+        val hashMap = HashMap<TrackingProtectionCategory, List<String>>()
+        for (item in list) {
+            when {
+                item.categories.contains(Tracker.Category.Cryptomining) -> hashMap[CRYPTOMINERS] =
+                    (hashMap[CRYPTOMINERS]
+                        ?: listOf()).plus(item.url.getHostFromUrl() ?: item.url)
+                item.categories.contains(Tracker.Category.Fingerprinting) -> hashMap[FINGERPRINTERS] =
+                    (hashMap[FINGERPRINTERS]
+                        ?: listOf()).plus(item.url.getHostFromUrl() ?: item.url)
+                item.categories.contains(Tracker.Category.Social) -> hashMap[SOCIAL_MEDIA_TRACKERS] =
+                    (hashMap[SOCIAL_MEDIA_TRACKERS] ?: listOf()).plus(
+                        item.url.getHostFromUrl() ?: item.url
+                    )
+                item.categories.contains(Tracker.Category.Content) -> hashMap[TRACKING_CONTENT] =
```

Comment:
I think `TRACKING_CONTENT` should also include `Ad`, `Analytic` and `Social`.

> If you identify an item that is CLASSIFIED_TRACKING, CLASSIFIED_TRACKING_AD, CLASSIFIED_TRACKING_ANALYTICS, CLASSIFIED_TRACKING_SOCIAL AND you have blocked it, count it as a Tracking Content. Return.

This could caused that the item will be on `SOCIAL_MEDIA_TRACKERS` and `TRACKING_CONTENT` at the same time. I think if one item is already in on a list above with higher hierarchy we shouldn't included. But we can as Tanvi to clarify in the document.